main.go

```diff
@@ -93,12 +93,17 @@ func main() {
 		os.Exit(1)
 	}
 
-	if err = (&controllers.VDOConfigReconciler{
+	vdoConfigReconciler := &controllers.VDOConfigReconciler{
 		Client:       mgr.GetClient(),
 		Logger:       ctrllog.Log.WithName("controllers").WithName("VDOConfig"),
 		Scheme:       mgr.GetScheme(),
 		ClientConfig: mgr.GetConfig(),
-	}).SetupWithManager(mgr); err != nil {
+	}
+
+	//Start configMap watcher for Compatibility Matrix
+	go vdoConfigReconciler.WatchForConfigMapChanges()
```

Comment:
why not just setup another watch when setting up the manager 
```

			Watches(
				&source.Kind{Type: &v1.ConfigMap{}},
				handler.EnqueueRequestsFromMapFunc(func(object client.Object) []reconcile.Request {
					cm, _ := object.(*v1.ConfigMap)
					if cm.Namespace == VDO_NAMESPACE && cm.Name == "test-cm" {
						
					}
					return nil
				}))
```
we expect there to be only vdoconfig resource in the cluster. Hence here in the watch we can fetch the vdo config resource and raise a reconcile request for the resource.
we should be leveraging the constructs in controller-runtime libraries to achieve these functionalities
we should also always compare the actual vs desired and perform reconcile to achieve the desired function